### ISSUE TITLE
RandomGamma turns negative values to NaN

### ISSUE THREAD

**Issue by fepegar** [fepegar/torchio#306]
I will elaborate later, but I think it's related to this the issue: https://stackoverflow.com/questions/45384602/numpy-runtimewarning-invalid-value-encountered-in-power

**Comment by fepegar** [comment]
@GFabien

**Comment by fepegar** [comment]
```python
In [14]: x
Out[14]:
tensor([ 0.0574, -0.2568,  0.3255,  0.0427, -0.2941, -0.2175, -0.2347, -0.4489,
        -0.0622, -0.4478])

In [15]: x ** 1.2
Out[15]:
tensor([0.0324,    nan, 0.2600, 0.0227,    nan,    nan,    nan,    nan,    nan,
           nan])
```

**Comment by GFabien** [comment]
That seems fair enough. I'm not sure how we should deal with this, introducing complex numbers is probably not a good idea. Doing as advised in the stackoverflow's answer sounds reasonable but it has to be mentioned in the doc. What do you think?

**Comment by fepegar** [comment]
What about the SO answer suggesting `np.sign(a) * (np.abs(a)) ** (1 / 3)`? What needs to be mentioned in the doc?

**Comment by GFabien** [comment]
> What about the SO answer suggesting np.sign(a) * (np.abs(a)) ** (1 / 3)? What needs to be mentioned in the doc?

Sorry I was referring to this answer. Even if it's natural it's not obvious that it is what the transform will do so maybe it's better to mention it in the doc.

**Comment by fepegar** [comment]
I see what you mean. Here's some more info: https://math.stackexchange.com/questions/317528/how-do-you-compute-negative-numbers-to-fractional-powers

I'm not sure what the best option would be. If we implement it like that, maybe that should indeed be explained in the docs **and** a `Warning` should be raised.

**Comment by GFabien** [comment]
> I'm not sure what the best option would be. If we implement it like that, maybe that should indeed be explained in the docs and a Warning should be raised.

I agree, if we think about contrast I think it makes more sense to not raise `np.sign(a)` at any power. This also avoids checking for fractional power to retrieve the power to which raise `np.sign(a)` when it is legal to do so (as explained in https://math.stackexchange.com/questions/317528/how-do-you-compute-negative-numbers-to-fractional-powers).

**Comment by fepegar** [comment]
I'm not sure what you mean. It's the `abs` that's raised, not the `sign`, right?

I've tried with gamma 1.28, here are some results for reference:

Original
![Screenshot](https://user-images.githubusercontent.com/12688084/93597994-e7b3bd80-f9b3-11ea-92e0-b18b1514ecf5.png)

After rescale (0, 1)
![Screenshot_2](https://user-images.githubusercontent.com/12688084/93598027-f7cb9d00-f9b3-11ea-959e-9f66fadd9899.png)

After rescale (-0.5, 0.5)
![Screenshot_3](https://user-images.githubusercontent.com/12688084/93598041-0023d800-f9b4-11ea-906b-2e3b81197869.png)

**Comment by GFabien** [comment]
> I'm not sure what you mean. It's the abs that's raised, not the sign, right?

Yes.

**Comment by fepegar** [comment]
Fixed in `v0.17.39`.
